RecoMET/METProducers/src/HcalNoiseInfoProducer.cc

```diff
@@ -500,9 +753,134 @@ HcalNoiseInfoProducer::filldigis(edm::Event& iEvent, const edm::EventSetup& iSet
+         // in the case that our stringent check below doesn't work 
+         // store the latest capID that has a match
+         int latest_cap_match = -1;
+         // reverse loop over TSs
+         while( start_ts >= 0 ) {
+           int ncheck = 0; // count the number of TS that were checked
+           int nmatch_cap = 0; // count the number of TS where capID matched
+           int nmatch_adc = 0; // count the number of TS where ADC matched
+
+           for( int cidx = start_ts, nidx = 0; cidx <= last_ts; cidx++, nidx++ ) { 
+             ncheck++;
+             // if we get an invald value, move on
+             if( lasmon_capids[fidx][cidx] == -1 ) continue;
+
+             if( lasmon_capids[fidx][cidx] == lasmon_capids[fidx+1][nidx] ) {
+               nmatch_cap++;
+             }
+             // check the data values as well
+             if( lasmon_adcs[fidx][cidx] == lasmon_adcs[fidx+1][nidx] ) {
+               nmatch_adc++;
+             }
+           }
+           bool cap_match = (ncheck == nmatch_cap);
+           bool adc_match = (ncheck == nmatch_adc);
+           if( cap_match && start_ts > latest_cap_match ) {
+             latest_cap_match = start_ts;
+           }
+           if( cap_match && adc_match ) {
+             // end the loop and we'll take the current start_ts
+             // as the end of the data for this fiber
+             break;
+           }
+           else {
+             // if we don't have a match, then decrement the 
+             // starting TS and check again
+             start_ts--;
+           }
+         }
+
+         // now make some sanity checks on the determined overlap index
+
+         if( start_ts == -1 ) {
+           // if we didn't find any match, use the capID only to compare
+           if( latest_cap_match < 0 ) {
+             //this shouldn't happen, in this case use all the data from the fiber
+             start_ts = lasmon_capids[fidx].size();
+           }
+           else {
+             // its possible that the timing of the fibers
+             // is shifted such that they do not overlap
+             // and we just want to stitch the fibers
+             // together with no removal.
+             // In this case the capIDs will match at the
+             // N-4 spot (and the ADCs will not)
+             // if this is not the case, then we just take
+             // the value of latest match
+             if( latest_cap_match == (last_ts - 3) ) {
+               start_ts = lasmon_capids[fidx].size();
+             } else {
+               start_ts = latest_cap_match;
+             }
+           }
+         }
+
+         // now store as the matching index
+         matching_idx[fidx] = start_ts;
+       }
+
+       // for the last fiber we always use all of the data
+       matching_idx[nFibers - 1] = 10;
+
+       // now loop over the time slices of each fiber and make the sum
+       int icombts = -1;
```

Comment:
I think that laserMonitorTSStart_ and laserMonitorTSEnd_ are meant to bound the time slices inside each fiber: if so, this initialization should be moved inside the first loop below.

Reply:
The aim of TSStart and TSEnd are to provide boundaries (if needed) for the charge integration after the data from the various fibers have been stitched together.   Therefore icombts should count from 0 to the total number of TS (which could be ~60)  and the limits evaluated based on that count

Reply:
Thank you for having checked and told.
The logic is not straightforward to follow, thugh... But ok, if you confirn that it is correct as such this is fine.